lib/models/os.coffee

```diff
@@ -42,26 +42,38 @@ BALENAOS_VERSION_REGEX = /v?\d+\.\d+\.\d+(\.rev\d+)?((\-|\+).+)?/
 
 DEVICE_TYPES_ENDPOINT_CACHING_INTERVAL = 10 * 60 * 1000 # 10 minutes
 
-withDeviceTypesEndpointCaching = (fn) ->
-	promiseMemoize(fn, { maxAge: DEVICE_TYPES_ENDPOINT_CACHING_INTERVAL })
-
 getOsModel = (deps, opts) ->
-	{ request } = deps
+	{ request, sdkInstance: { auth } } = deps
 	{ apiUrl, isBrowser } = opts
 
 	configModel = once -> require('./config')(deps, opts)
 	applicationModel = once -> require('./application')(deps, opts)
 
-	getDeviceTypes = once ->
+	withDeviceTypesEndpointCaching = (fn) ->
```

Comment:
Why do we memoize the function and not the result from it, if it succeeds?

Reply:
The package used here already caches the result only when the promise gets resolved.
We do follow the same pattern in other top components as well and it saves us from
* manually denouncing calls while there is already one ongoing, in which case it returns the same ongoing promise
* having a manual `If (cached) { return cached; }`

Is there a reason that you think we should be in favor of doing the above two on our own code?